app/models/acts_as_customizable/calculated_value.rb

```diff
@@ -42,28 +42,58 @@ def enabled_custom_field_ids
     def calculate_custom_fields(custom_fields)
       return if custom_fields.empty?
 
-      unless custom_fields.all?(&:field_format_calculated_value?)
-        fail ArgumentError,
-             "Expected array of calculated value custom fields"
-      end
+      validate_custom_fields_for_calculation!(custom_fields)
 
       enabled_ids = enabled_custom_field_ids
+      given = calculated_value_fields_given(custom_fields:, enabled_ids:)
+
+      remove_calculated_value_errors!(custom_fields.map(&:id))
 
-      result = calculate_custom_fields_result(
-        given: calculated_value_fields_given(custom_fields:, enabled_ids:),
+      result, errors = calculate_custom_fields_result(
+        given:,
         to_compute: calculated_value_fields_to_compute(custom_fields:, enabled_ids:)
-      )
+      ).values_at(:result, :errors)
```

Comment:
Pattern matching variant, I don't think I saw it used in the wild, so not sure how readable is it
```suggestion
      calculate_custom_fields_result(
        given:,
to_compute: calculated_value_fields_to_compute(custom_fields:, enabled_ids:)
      ) => { result:, errors: }
```

Reply:
Haven't seen that before. It's a bit obscure, but elegant. I like it! :shipit: